src/path.c

```diff
@@ -865,6 +933,116 @@ ly_path_compile_predicate_leafref(const struct lysc_node *ctx_node, const struct
     return (ret == LY_ENOTFOUND) ? LY_EVALID : ret;
 }
 
+/**
+ * @brief Appeds path elemets from source to destination array
+ *
+ * @param[in] ctx libyang context.
+ * @param[in] src The source path
+ * @param[in,out] dst The destination path
+ * @return LY_ERR value.
+ */
+static LY_ERR
+ly_path_append(const struct ly_ctx *ctx, const struct ly_path *src, struct ly_path **dst)
+{
+    LY_ERR ret = LY_SUCCESS;
+    struct ly_path *p;
+    size_t u;
+
+    LY_ARRAY_FOR(src, u) {
+        LY_ARRAY_NEW_GOTO(ctx, *dst, p, ret, cleanup);
+        p->node = src[u].node;
+        p->ext = src[u].ext;
```

Comment:
While most likely not needed for this specific use-case, but it may be required eventually and the function pretends to be generic, so please add predicate duplication.

Reply:
Done